functions/src/custom/vanity.ts

```diff
@@ -47,3 +57,102 @@ export const create_vanity_link = functions.firestore
       Sentry.captureException(error);
     }
   });
+
+export const build_vanity_link = functions.firestore
+  .document("typeform/{document_name}")
+  .onCreate(async (snap, context) => {
+    const document = snap.data();
+    try {
+      if (document.typeform_id == "Link Generator") {
+        const typeform_results = document.data;
+        let full_name = "";
+        let email = "";
+        let destination = "";
+        let primary_domain = "";
+        let subdomain = "";
+        let slashtag = "";
+
+        typeform_results.forEach((element: any) => {
+          const fullname_question = "full name";
+          const email_question = "email";
+          const destination_question = "vanity link";
+          const primary_domain_question = "primary domain";
+          const subdomain_question = "vanity domain";
+          const slashtag_question = "slashtag";
+          if (element.question.includes(fullname_question)) {
+            full_name = element.answer;
+          }
+          if (element.question.includes(email_question)) {
+            email = element.answer;
+          }
+          if (element.question.includes(destination_question)) {
+            destination = element.answer;
+          }
+          if (element.question.includes(primary_domain_question)) {
+            primary_domain = element.answer.label;
+          }
+          if (element.question.includes(subdomain_question) && element.answer.label !== "") {
+            subdomain = element.answer.label;
+          }
+          if (element.question.includes(slashtag_question)) {
+            slashtag = element.answer;
+          }
+        });
+
+        const data: vanity = {
+          destination: destination,
+          primary_domain: primary_domain,
+          subdomain: subdomain,
+          slashtag: slashtag,
+        };
+        create_link(data);
+        send_confirmation(data, email, full_name);
+        console.log(full_name, email);
+      }
```

Comment:
```suggestion
      if (document.typeform_id !== "Link Generator") {
        return;
      }
      const typeform_results = document.data;
      let full_name = "";
      let email = "";
      let destination = "";
      let primary_domain = "";
      let subdomain = "";
      let slashtag = "";

      typeform_results.forEach((element: any) => {
        const fullname_question = "full name";
        const email_question = "email";
        const destination_question = "vanity link";
        const primary_domain_question = "primary domain";
        const subdomain_question = "vanity domain";
        const slashtag_question = "slashtag";
        if (element.question.includes(fullname_question)) {
          full_name = element.answer;
        }
        if (element.question.includes(email_question)) {
          email = element.answer;
        }
        if (element.question.includes(destination_question)) {
          destination = element.answer;
        }
        if (element.question.includes(primary_domain_question)) {
          primary_domain = element.answer.label;
        }
        if (element.question.includes(subdomain_question) && element.answer.label !== "") {
          subdomain = element.answer.label;
        }
        if (element.question.includes(slashtag_question)) {
          slashtag = element.answer;
        }
      });
     const data: vanity = {
        destination: destination,
        primary_domain: primary_domain,
        subdomain: subdomain,
        slashtag: slashtag,
      };
      create_link(data);
      send_confirmation(data, email, full_name);
      console.log(full_name, email);
    }
```

Instead of a large `if` block with no else, flatten the code by checking for the condition's opposite. Also, use triple equals.